patternfly/patternfly-elements
# fix(create): fix templates for single-package repos

## What I did

- Generator now provides the option to choose the css flavour for the element, e.g. `scss`, `css`, or `postcss`
- Generator templates for single-package repos (i.e. RHDS) are fixed. internally, generator hardcodes some values for RHDS, this can be updated later as it becomes an issue

## Testing Instructions
- Confirm that the generator works in pfe
```bash
npm run build -w @patternfly/create-element
```
```bash
npm run new
```
```bash
npm start
```
Confirm that the new element works
- Confirm that the generator works in rhds
```bash
cd ../path/to/red-hat-design-system
```
It will be necessary to copy over some of create's dependencies to the rhds node_modules folder.
You may need to update these paths to suit your local setup
```bash
rm -rf node_modules/@patternfly/create-element/                                           
cp -R ../../patternfly/patternfly-elements/tools/create-element/ node_modules/@patternfly/create-element
cp -R ../../patternfly/patternfly-elements/node_modules/execa/ node_modules/execa
cp -R ../../patternfly/patternfly-elements/node_modules/npm-run-path/ node_modules/npm-run-path 
npx @patternfly/create-element -n rh-jazz-hands --overwrite
```
